src/astro/databases/snowflake.py

```diff
@@ -57,6 +76,252 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
         qualified_name = ".".join(name for name in qualified_name_lists if name)
         return qualified_name
 
+    def check_optimised_path(self, source_file: File, target_table: Table) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        """
+        return bool(self.native_support.get(source_file.location.location_type))
+
+    def optimised_transfer(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> None:
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+
+        :param source_file: Source file
+        :param target_table: Output target table on snowflake
+        :param if_exists: Update strategy for file
+        :param kwargs:
+        """
+        method_name = self.native_support.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            try:
+                if transfer_method(
+                    source_file=source_file,
+                    target_table=target_table,
+                    if_exists=if_exists,
+                    **kwargs,
+                ):
+                    return
+            except Exception as exec_err:
+                raise exec_err
+        return
+
+    @staticmethod
+    def get_gcs_project_id_from_conn(conn: connection) -> str:
```

Comment:
Where are we using this method? Does it receive a `file_conn` or a `db_conn`?

Reply:
I was planing to get the chunk of file in using GCP connection and fetch first two to detect schema for CSV. I am implementing that part.

Reply:
Removed this. Using pandas to do that.